Added method to check namelist errors

daveh150 opened the pull request:
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>


## PR checklist
- [ x] Short (1 sentence) summary of your PR: 
    Provide specific namelist line errors
- [ x] Developer(s): 
    David Hebert
- [ x] Suggest PR reviewers from list in the column to the right.
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
     Quick Suite BFB test on GAFFNEY:
21 measured results of 21 total results
21 of 21 tests PASSED
0 of 21 tests PENDING
0 of 21 tests MISSING data
0 of 21 tests FAILED

- How much do the PR code changes differ from the unmodified code? 
    - [ x] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ x] Yes
    - [ ] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [ x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [ ] Yes
    - [ x] No, does the documentation need to be updated at a later time?
        - [x ] Yes
        - [ ] No 
- [ ] Please provide any additional information or relevant details below:
When namelist has incorrect variable name, the error only lists the namelist. This addition will tell the namelist line that was read incorrectly. This will help when namelist variable names are updated but you didn't change your ice_in file.

apcraig reviewed (changes requested):
Suggest we add the colon in the error message like in CICE.  Also, check for that extra space at the end of 'error reading namelist '.  At least one is missing.

daveh150 commented:
I updated the error messages to be consistent with the ones in CICE - made output 'ERROR:' 

Also added extra space after error message in icedrv_init.F90

eclare108213 reviewed (approved):
Thanks for this change -- I have struggled to debug these kinds of errors, and this will definitely help.
